### ISSUE TITLE
[M1] data: Audit and prepare all available retinal datasets

### ISSUE THREAD

**Issue by Devrajj-Singh** [Devrajj-Singh/NeuroVista-DR#20]
## Objective

Establish a reproducible multi-dataset data foundation for the full NeuroVista-DR development cycle before large-scale GPU training.

At the current stage, **APTOS 2019 and IDRiD are the datasets actually available in the repository**. The other planned dataset folders must not be treated as available until their data has been obtained and legally/technically verified.

Only APTOS has been used for the existing DR classification baseline. IDRiD has not yet been integrated into classifier training.

This issue replaces the prototype assumption of using one dataset and prepares the data layer for full-scale model development.

## Current Dataset State

| Dataset | Current state | Planned role |
|---|---|---|
| APTOS 2019 | Available; used for baseline | Primary DR grading development |
| IDRiD | Available; not yet integrated into classifier training | DR grading + lesion segmentation/evidence |
| DeepDRiD | Folder exists but data unavailable | Image-quality modeling |
| EyePACS | Folder exists but data unavailable | Backbone pretraining / robustness |
| Messidor-2 | Folder exists but data unavailable | External validation only |

Do not mark unavailable datasets as prepared or used until the actual data is present and validated.

### APTOS 2019

- [ ] Reconfirm image/label integrity
- [ ] Preserve the existing duplicate/leakage checks
- [ ] Review current train/validation split
- [ ] Create a formal held-out test strategy
- [ ] Record dataset metadata and provenance

### IDRiD

- [ ] Verify all image files are present
- [ ] Verify all image-level DR grades 0–4
- [ ] Parse lesion masks
- [ ] Parse optic-disc/fovea localization annotations
- [ ] Link image-level and pixel-level annotations correctly
- [ ] Verify train/test organization supplied by dataset
- [ ] Check image integrity and missing annotations
- [ ] Document license/provenance
- [ ] Create machine-readable metadata

### Planned / Currently Unavailable Datasets

For DeepDRiD, EyePACS, and Messidor-2:

- [ ] Verify whether the dataset can legally be obtained/used
- [ ] Add download/setup instructions without committing restricted raw data
- [ ] Validate only after the actual dataset is available
- [ ] Document intended role before integration

## Unified Metadata

Create a consistent metadata representation containing, where available:

```text
image_path
dataset
patient_id
eye_id
dr_grade
annotation_type
split
```

For lesion annotations, maintain separate references rather than flattening masks into the classification labels.

## Leakage Prevention

- [ ] Use patient-level separation whenever patient identifiers are available
- [ ] Avoid cross-dataset contamination between training and external validation
- [ ] Check duplicate/near-duplicate images where practical
- [ ] Keep Messidor-2 completely isolated from training and tuning if it becomes available

## Output

Create/update:

```text
data/
├── raw/
├── processed/
│   ├── aptos_metadata.csv
│   ├── idrid_metadata.csv
│   └── unified_metadata.csv
└── README.md
```

Do not commit restricted raw datasets to GitHub.

## Acceptance Criteria

- [ ] APTOS data state documented and reproducible
- [ ] IDRiD fully inspected and metadata generated
- [ ] IDRiD lesion annotations are accessible through the data pipeline
- [ ] Unavailable datasets are explicitly marked unavailable
- [ ] Unified metadata schema exists
- [ ] Split/leakage policy documented
- [ ] Dataset provenance/licenses documented
- [ ] Data pipeline can be consumed by future training code
- [ ] No raw restricted dataset files are committed to the repository

## Dependencies

This issue should be completed before the full-scale GPU training issue.
